### ISSUE TITLE
Service "Not loaded" v4.0.38

### ISSUE THREAD

**Issue by rholligan** [BJReplay/ha-solcast-solar#85]
### The problem

I updated this morning to v4.0.38, and restarted HA as prompted. Now the Solcast integration appears as "not loaded" 
![image](https://github.com/user-attachments/assets/4b1e12dd-fe99-4cfb-a80c-143c5a27afe1)

I also get this log entry:

```
Log Details (ERROR)
Logger: homeassistant.setup
Source: setup.py:271
First occurred: 09:38:36 (1 occurrences)
Last logged: 09:38:36

Setup failed for custom integration 'solcast_solar': Requirements for solcast_solar not found: ['scipy>=1.0.0'].
```
I noticed the mention of "scipy", which also appears elsewhere in my logs:

```
Logger: homeassistant.util.package
Source: util/package.py:123
First occurred: 09:34:33 (3 occurrences)
Last logged: 09:38:36

Unable to install package scipy>=1.0.0: error: subprocess-exited-with-error × Preparing metadata (pyproject.toml) did not run successfully. │ exit code: 1 ╰─> [34 lines of output] Traceback (most recent call last): File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module> main() File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main json_out['return_val'] = hook(**hook_input['kwargs']) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 152, in prepare_metadata_for_build_wheel whl_basename = backend.build_wheel(metadata_directory, config_settings) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-ewt7ypug/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 1020, in wrapper return func(*args, **kwargs) ^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-ewt7ypug/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 1073, in build_wheel with _project(config_settings) as project: File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__ return next(self.gen) ^^^^^^^^^^^^^^ File "/tmp/pip-build-env-ewt7ypug/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 947, in _project yield Project(source_dir, build_dir, meson_args, editable_verbose) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-ewt7ypug/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 662, in __init__ self._meson = _get_meson_command(pyproject_config.get('meson')) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-ewt7ypug/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 982, in _get_meson_command meson_version = subprocess.run(cmd + ['--version'], check=False, text=True, capture_output=True).stdout ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/subprocess.py", line 548, in run with Popen(*popenargs, **kwargs) as process: ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/subprocess.py", line 1026, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "/usr/local/lib/python3.12/subprocess.py", line 1955, in _execute_child raise child_exception_type(errno_num, err_msg, err_filename) PermissionError: [Errno 13] Permission denied: 'meson' [end of output] note: This error originates from a subprocess, and is likely not a problem with pip. error: metadata-generation-failed × Encountered error while generating package metadata. ╰─> See above for output. note: This is an issue with the package mentioned above, not pip. hint: See above for details.
Unable to install package scipy>=1.0.0: error: subprocess-exited-with-error × Preparing metadata (pyproject.toml) did not run successfully. │ exit code: 1 ╰─> [34 lines of output] Traceback (most recent call last): File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module> main() File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main json_out['return_val'] = hook(**hook_input['kwargs']) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 152, in prepare_metadata_for_build_wheel whl_basename = backend.build_wheel(metadata_directory, config_settings) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-trbe6431/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 1020, in wrapper return func(*args, **kwargs) ^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-trbe6431/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 1073, in build_wheel with _project(config_settings) as project: File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__ return next(self.gen) ^^^^^^^^^^^^^^ File "/tmp/pip-build-env-trbe6431/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 947, in _project yield Project(source_dir, build_dir, meson_args, editable_verbose) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-trbe6431/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 662, in __init__ self._meson = _get_meson_command(pyproject_config.get('meson')) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-trbe6431/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 982, in _get_meson_command meson_version = subprocess.run(cmd + ['--version'], check=False, text=True, capture_output=True).stdout ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/subprocess.py", line 548, in run with Popen(*popenargs, **kwargs) as process: ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/subprocess.py", line 1026, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "/usr/local/lib/python3.12/subprocess.py", line 1955, in _execute_child raise child_exception_type(errno_num, err_msg, err_filename) PermissionError: [Errno 13] Permission denied: 'meson' [end of output] note: This error originates from a subprocess, and is likely not a problem with pip. error: metadata-generation-failed × Encountered error while generating package metadata. ╰─> See above for output. note: This is an issue with the package mentioned above, not pip. hint: See above for details.
Unable to install package scipy>=1.0.0: error: subprocess-exited-with-error × Preparing metadata (pyproject.toml) did not run successfully. │ exit code: 1 ╰─> [34 lines of output] Traceback (most recent call last): File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module> main() File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main json_out['return_val'] = hook(**hook_input['kwargs']) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 152, in prepare_metadata_for_build_wheel whl_basename = backend.build_wheel(metadata_directory, config_settings) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-3cnmixjj/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 1020, in wrapper return func(*args, **kwargs) ^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-3cnmixjj/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 1073, in build_wheel with _project(config_settings) as project: File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__ return next(self.gen) ^^^^^^^^^^^^^^ File "/tmp/pip-build-env-3cnmixjj/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 947, in _project yield Project(source_dir, build_dir, meson_args, editable_verbose) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-3cnmixjj/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 662, in __init__ self._meson = _get_meson_command(pyproject_config.get('meson')) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/tmp/pip-build-env-3cnmixjj/overlay/lib/python3.12/site-packages/mesonpy/__init__.py", line 982, in _get_meson_command meson_version = subprocess.run(cmd + ['--version'], check=False, text=True, capture_output=True).stdout ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/subprocess.py", line 548, in run with Popen(*popenargs, **kwargs) as process: ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/subprocess.py", line 1026, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "/usr/local/lib/python3.12/subprocess.py", line 1955, in _execute_child raise child_exception_type(errno_num, err_msg, err_filename) PermissionError: [Errno 13] Permission denied: 'meson' [end of output] note: This error originates from a subprocess, and is likely not a problem with pip. error: metadata-generation-failed × Encountered error while generating package metadata. ╰─> See above for output. note: This is an issue with the package mentioned above, not pip. hint: See above for details.
```

### What version of Solcast Integration has the issue?

v4.0.38

### What version of Home Assistant are you using?

HA Core 2024.7.2

### Diagnostics information

Not possible

### Logs debug/info/error output

```txt
See above
```

**Comment by autoSteve** [comment]
Interesting, @rholligan.

Does your HA have Internet access? Probably.

Could I get you to please re-upgrade to 4.0.38 (assuming that you've downgraded by now), and change scipy in manifest.json to >=1.9.0.

There are 45 other downloads so far with no reported issue, so this surprises me.

**Comment by rholligan** [comment]
Yes it does have internet access. 

I will try the steps you mentioned and report back.

**Comment by autoSteve** [comment]
Further, try adding "numpy>=1.21.4" as well.

**Comment by ThomDietrich** [comment]
I am seeing the same error. Just confirming this is not a single system issue.

**Comment by autoSteve** [comment]
(You did completely restart HA @rholligan, yes? Just confirming...)

This is a serious problem if two people so far have reported hitting the issue, which is 5% of current downloads, and it doesn't sound easily solved by fiddling the manifest. I'm also on 2024.7.2, and nothing blew up here. @isorin ran the alpha too without complications, presumably @gcoan too.

Downgrade to 4.0.37 and I'll stare at it a bit more.

**Comment by rholligan** [comment]
> (You did completely restart HA @rholligan, yes? Just confirming...)
> 
> This is a serious problem if two people so far have reported hitting the issue, which is 5% of current downloads, and it doesn't sound easily solved by fiddling the manifest. I'm also on 2024.7.2, and nothing blew up here. @isorin ran the alpha too without complications, presumably @gcoan too.
> 
> Downgrade to 4.0.37 and I'll stare at it a bit more.

Yes I did. Will revert to 4.0.37 for now

**Comment by Gavin-A92** [comment]
Just here to add a "me too" to the issue (so you can monitor occurances)......running Core 2024.7.2 with HAOS on Pi5 and seeing the unable to install scipy error entries in the logs as mentioned above.

**Comment by autoSteve** [comment]
Thanks for that. I'm thinking of pulling down v4.0.38 unless I can come up with something reasonably quickly.

It's why we have betas...

**Comment by autoSteve** [comment]
Is everyone hitting this running on a RPi?

**Comment by rholligan** [comment]
I am running a Home Assistant Green

**Comment by cncservsys** [comment]
I also have this issue - Running RPi

**Comment by ThomDietrich** [comment]
Hardkernel ODROID-N2+
Core 2024.7.2
Supervisor 2024.06.2
Operating System 12.4
Frontend 20240710.0

**Comment by gcoan** [comment]
I'm on HAOS as a windows VM, 12.4, HA v2024.7.2, running fine

**Comment by Thrasher2020** [comment]
Same issue here and using the new HACS so downgrading isn't as easy any more :(

**Comment by jeffb1974** [comment]
Same issue, also on RPI.  Went back to 4.0.37 and all OK.

**Comment by autoSteve** [comment]
In the next release (most likely) the use of scipy/numpy has been retired. The last thing I want to see is less reliability, so old school math for the win.

Some of the other changes that I am proposing for that release are very much worthy of an alpha test, as they add new configuration capabilities. This math change will also be in the payload of that alpha. If anyone would like to participate, round up the discussion topic that I will create in a little while, once my testing is done.

**Comment by ladaan** [comment]
I have upgraded to v4.0.37 yesterday (had similar issue as other with scipy module on v4.0.38). However, I miss the dashed line, the prediction. Where should I check, troubleshoot? Does it work for others?

**Comment by autoSteve** [comment]
An alpha of v4.0.39 is up. See https://github.com/BJReplay/ha-solcast-solar/discussions/87.

**Comment by autoSteve** [comment]
> However, I miss the dashed line, the prediction

Sounds like the integration isn't loading properly for you.

Got any logs for me @ladaan? Debug logs would be ace, but I'll take anything.

**Comment by autoSteve** [comment]
Noted on the 'similar scipy thing', but do feel free to open a separate issue @ladaan, including logs and the rest. Your issue may be distict.

**Comment by ladaan** [comment]
> Noted on the 'similar scipy thing', but do feel free to open a separate issue @ladaan, including logs and the rest. Your issue may be distinct.

Issue on my side. Once the solcast was removed and added again from this repo, the solcast was not enabled in Energy dashboard, Solar Panels section.

**Comment by autoSteve** [comment]
Good to hear, @ladaan.

You didn't need to remove the integration. Just switch repos and re-download, and all config would have carried over.

**Comment by Godzillahu** [comment]
Hi,

I have the same problem. No data coming. I've already tried the alpha, but it doesn't receive data either. I just reinstalled v4.0.38.
![Screenshot from 2024-07-17 09-50-51](https://github.com/user-attachments/assets/2e0457b0-6a25-4031-b870-8c90c7800f88)

**Comment by autoSteve** [comment]
I'm blind from that info, @Godzillahu. I can't tell you much.

But what I can definitely tell you is that it is _not_ the same issue as this one if you installed the alpha v4.0.39. That code does not carry a dependency on scipi, so it _cannot_ be the same issue... The alpha as designed to _solve_ this issue, and nearly made my math brain explode in the process of building it.

Please open a new issue, and with debug logging enabled so you can include that detail so I can see what's really going on in your case.

**Comment by autoSteve** [comment]
Do I have any more takers to try v4.0.39-alpha1? There are five downloads (including myself, @gcoan and @Godzillahu) that have not encountered anything untoward.

It is looking like a winner, but I'd really like more experience with it.

The scipy/math stuff from v4.0.38 is solved, with forecast remaining today values solid for me, and tracking lovely spline interpolations, and I'm loving the new options to suppress sensor attributes that I don't need.

Yes, it's a bit of a palaver to switch repos to @autoSteve to download, but quite straightforward and low risk really. All revealed in https://github.com/BJReplay/ha-solcast-solar/discussions/87.

After the alpha is declared a success switching back is equally a "palaver", but again not hard, and no data loss.

Any takers?

Do make yourselves known in the discussion at https://github.com/BJReplay/ha-solcast-solar/discussions/87 so you'll know when the alpha ends and it transitions to beta.

I honestly should have sent this as v4.0.39-alpha1 to BJReplay in retrospect, given its stability, but did not want to potentially pollute that code given the payload.

**Comment by ladaan** [comment]
I have followed instructions and the v4.0.39-alpha1 broke it.
Below is what I can see in the log.

Logger: homeassistant.config_entries
Source: config_entries.py:586
First occurred: 16:46:37 (1 occurrences)
Last logged: 16:46:37

Error setting up entry Solcast Solar for solcast_solar
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 586, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solcast_solar/__init__.py", line 135, in async_setup_entry
    entry.options[BRK_ESTIMATE],
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^
KeyError: 'attr_brk_estimate'

**Comment by autoSteve** [comment]
Brilliant @ladaan. Do you have a debug log to also share from that happening? `less /config/homeassistant.log`, keyboard: `/solcast', hunt for a line that is upgrading the config version?

**Comment by autoSteve** [comment]
I think I see what is going on. Expect v4.0.39-alpha2...

**Comment by ladaan** [comment]
Actually, there is no more info in the log than I shared.

**Comment by autoSteve** [comment]
v4.0.39-alpha2 published at autoSteve, @ladaan.

**Comment by autoSteve** [comment]
(Go to HACS, find 'Solcast', three-dots menu, update information, redownload with reveal alpha2.

**Comment by ladaan** [comment]
Now it works. Thank you.

**Comment by jeffb1974** [comment]
Alpha 2 works for me also (RPI).  Alpha1 did not work.

Thanks!

**Comment by rholligan** [comment]
> Do I have any more takers to try v4.0.39-alpha1? There are five downloads (including myself, @gcoan and @Godzillahu) that have not encountered anything untoward.
> 
> It is looking like a winner, but I'd really like more experience with it.
> 
> The scipy/math stuff from v4.0.38 is solved, with forecast remaining today values solid for me, and tracking lovely spline interpolations, and I'm loving the new options to suppress sensor attributes that I don't need.
> 
> Yes, it's a bit of a palaver to switch repos to @autoSteve to download, but quite straightforward and low risk really. All revealed in #87.
> 
> After the alpha is declared a success switching back is equally a "palaver", but again not hard, and no data loss.
> 
> Any takers?
> 
> Do make yourselves known in the discussion at #87 so you'll know when the alpha ends and it transitions to beta.
> 
> I honestly should have sent this as v4.0.39-alpha1 to BJReplay in retrospect, given its stability, but did not want to potentially pollute that code given the payload.

I'm going to give it a try today - will let you know

**Comment by autoSteve** [comment]
Great work, GitHub.

What I said over in #87 is...

FYI, v4.0.39 draft release is primed and ready, which will end the alpha, and switch back to the @BJReplay repo. More to come.

Unless anyone has grievances to report...

Given the whole CrowdStrike thing you'll forgive me for delaying for about ten hours while I sleep and then double check all work with fresh eyes... You Europeans will be tucked up in bed by that time, so it'll probably be there when you wake up much later...

Reverting to the BJReplay repo in HACS is straightforward, as noted in the original post #87, except that will download v4.0.31. Choose to re-download, and select v4.0.39 after I pull the trigger. Then restart HA.

Lemme know if there are any issues doing so.

I'll advise here when v4.0.39 is up at BJReplay.

BTW, I will be "out-of-office" for a week come tomorrow. I'm heading to Tokyo to rubberneck at stuff, and hang out in odd bars, so responses may be delayed depending on how good the rubbernecking is. I'm tipping the rubbernecking will be epic. (Did someone say MARIO CARTS through the streets of Tokyo??? SEND IT!!!)

#84, #85, and more.

**Comment by autoSteve** [comment]
All, v4.0.39 is up.

To switch from @autoSteve, in HACS:

* Add the custom repo `https://github.com/BJReplay/ha-solcast-solar`
* Remove custom repo `https://github.com/autoSteve/dev-solcast-solar'
* Search for `solcast`, open it, download button at bottom right, which will get you back on v4.0.31
* Restart HA
* Back in hacs, re-download Solcast integration, turn on `Show beta versions`, select v4.0.39, download
* Restart HA
* Verify in logs that v4.0.39 is running

It may be possible to do this with one HA restart. I had some minor HACS weirdness, ending up on v4.0.38, so upgraded in the HA settings and restarted HA again.
